benchmarks/renderImageItem.py

```diff
@@ -22,117 +23,156 @@ def renderQImage(*args, **kwargs):
+    dtypes: list[npt.DTypeLike]
+    lut_dtypes: list[npt.DTypeLike]
+
+class TimeSuite:
+    unit = "seconds"
+    param_names = ["size", "acceleration", "use_levels", "dtype", "lut_dtype"]
+    params = Parameters(
+        [
+            (256, 256),
+            (512, 512),
+            (1024, 1024),
+            (2048, 2048),
+            (3072, 3072),
+            (4096, 4096)
+        ],                                      # size
+        ["numpy", "numba", "cupy"],
+        [True, False],                          # use_levels
+        ['uint8', 'uint16', 'float32'],         # dtype
+        ['uint8', None]                         # lut_dtype
+    )
     def __init__(self):
-        super(_TimeSuite, self).__init__()
-        self.size = None
-        self.float_data = None
-        self.uint8_data = None
-        self.uint8_lut = None
-        self.uint16_data = None
-        self.uint16_lut = None
-        self.cupy_uint16_lut = None
-        self.cupy_uint8_lut = None
-
-    def setup(self):
-        size = (self.size, self.size)
-        self.float_data, self.uint16_data, self.uint8_data, self.uint16_lut, self.uint8_lut = self._create_data(
-            size, np
-        )
-        if numba is not None:
+        self.data = np.empty((), dtype=np.uint8)
+        self.lut = np.empty((), dtype=np.ubyte)
+        self.levels = None
+
+    def teardown(self, *args, **kwargs):
+        # toggle options off
+        pg.setConfigOption("useNumba", False)
+        pg.setConfigOption("useCupy", False)
+
+    def setup_cache(self) -> dict:
+        accelerations = [np]
+        if cp is not None:
+            accelerations.append(cp)
+
+        cache = {}
+
+        for xp in accelerations:
+            cache[xp.__name__] = {"lut": {}, "data": {}}
+
+            # handle lut caching
+            c_map = xp.array([[-500.0, 255.0], [-255.0, 255.0], [0.0, 500.0]])
+            for lut_dtype in self.params.lut_dtypes:
+                if lut_dtype is None:
+                    continue
+                bits = xp.dtype(lut_dtype).itemsize * 8
+                # create the LUT
+                lut = xp.zeros((2 ** bits, 4), dtype="ubyte")
+                for i in range(3):
+                    lut[:, i] = xp.clip(xp.linspace(c_map[i][0], c_map[i][1], 2 ** bits), 0, 255)
+                lut[:, -1] = 255
+                cache[xp.__name__]["lut"][lut_dtype] = lut
+
+            # handle data caching
+            for dtype in self.params.dtypes:
+                cache[xp.__name__]["data"][dtype] = {}
+                for size in self.params.sizes:
+                    if xp.dtype(dtype) in (xp.float32, xp.float64):
+                        data = xp.random.default_rng().standard_normal(size=size, dtype=dtype)
+                    else:
+                        iinfo = xp.iinfo(dtype)
+                        data = xp.random.default_rng().integers(
+                            low=iinfo.min,
+                            high=iinfo.max,
```

Comment:
You need to add the argument `endpoint=True`.

Reply:
thank you!